### ISSUE TITLE
Pre-flight check for libvirt default network fails when LANG is set to Swiss-German

### ISSUE THREAD

**Issue by mwaeckerlin** [minishift/minishift#2121]
### General information

  * Minishift version: 1.14.0
  * OS: Linux, Ubuntu 16.04
  * Hypervisor: KVM and VirtualBox should be availablle, how do I choose one if them?!?

### Steps to reproduce

 1. Goto https://www.openshift.org/minishift/
 1. Follow link to https://github.com/minishift/minishift/releases
 1. Download latest minishift release
 1. Back to previous document, follow link to https://docs.openshift.org/latest/minishift/getting-started/installing.html
 1. In _Linux_, Follow link to https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html
 1. execute the commands: 
        `sudo curl -L https://github.com/dhiltgen/docker-machine-kvm/releases/download/v0.7.0/docker-machine-driver-kvm -o /usr/local/bin/docker-machine-driver-kvm`
        `sudo chmod +x /usr/local/bin/docker-machine-driver-kvm`
        `sudo apt install libvirt-bin qemu-kvm`
        `sudo usermod -a -G libvirtd marc`
       `newgrp libvirtd`
 1. Back to last page, look for next instuctions, after a while found it in the left navigation, click https://docs.openshift.org/latest/minishift/getting-started/quickstart.html
 1. execute `minishift start`

### Expected

Minishift starts without error.

### Actual

Aborts

### Logs

```
-- Starting profile 'minishift'
-- Checking if requested OpenShift version 'v3.7.1' is valid ... OK
-- Checking if requested OpenShift version 'v3.7.1' is supported ... OK
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... FAIL
   See the 'Setting Up the Driver Plug-in' topic (https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html) for more information
```

The given URL does not help,I already follwed all steps mentioned there.

### Notes

Opened new issue as requested by @gbraad  in #1998

BTW: Documentation is a mess!

My target: I would like to setup a OpenShift scenario in Virtual Box.

**Comment by anjannath** [comment]
@mwaeckerlin can you check if the default network is active, please use this command and paste the o/p here
`virsh --connect qemu:///system net-list`

**Comment by mwaeckerlin** [comment]
Yes it is:
```
marc@jupiter:/tmp$ LANG= virsh --connect qemu:///system net-list
 Name                 State      Autostart     Persistent
----------------------------------------------------------
 default              active     yes           yes
```

**Comment by mwaeckerlin** [comment]
BTW: Is it possible to get a useful error message from this black box? What command exactly failed?

Perhaps you should know, that my notebook is running in a company environment, where many networks are already used, i.e. the one OpenShift installes the registry. So I expect minishift to create a routing to it's own independent NAT translated network?

**Comment by praveenkumar** [comment]
@mwaeckerlin Following is the code which check the default network state. as per your output of `virsh` command it looks active but this check is failing. there is also a way to ignore that test by executing `minishift config set skip-check-kvm-driver true` and then run again.

```golang
//checkLibvirtDefaultNetworkActive returns true if the "default" network is active
func checkLibvirtDefaultNetworkActive() bool {
	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-list")
	cmd.Env = cmdUtils.ReplaceEnv(os.Environ(), "LC_ALL", "C")
	stdOutStdError, err := cmd.CombinedOutput()
	if err != nil {
		return false
	}
	stdOut := fmt.Sprintf("%s", stdOutStdError)
	outputSlice := strings.Split(stdOut, "\n")

	for _, stdOut = range outputSlice {
		stdOut = strings.TrimSpace(stdOut)
		match, err := regexp.MatchString("^default\\s", stdOut)
		if err != nil {
			return false
		}
		if match && strings.Contains(stdOut, "active") {
			return true
		}
	}
	return false
}
```

**Comment by gbraad** [comment]
can we keep a constructive tone?

please explain what makes you say, the documentation is a mess. We have an issue already, being worked on #1817 to streamline the installation descriptions.
Also, calling something bad programming is generally a no-go. We try to set `cmd.Env = cmdUtils.ReplaceEnv(os.Environ(), "LC_ALL", "C")`, but for some reason this fails on your machine? We have tested this under several circumstances and this is an identical fix (although simplified) as provided by an outside contributor. So, in your case `LANG` is also set. OK, we will add a check for this and will continue on #1584 and related tasks for Linux and macOS. However, at this stage we haven't had bandwidth to handle all of this at once.

**Comment by gbraad** [comment]
> My target: I would like to setup a OpenShift scenario in Virtual Box.

in that case, use `--vm-driver virtualbox`.

**Comment by gbraad** [comment]
Off-topic: a PR for the documentation is available as #2025

**Comment by mwaeckerlin** [comment]
@gbraad, Sorry, I criticize with the idea of helping to make things better. I thought my problems regarding documentation was obvious from my description. Reading was not so straight-forward, so at the end, when the script did not work as expected, I had no idea whether it because of a bug in the script or because I missed something in the documentation.

Starting at https://www.openshift.org/minishift/, the links to xhyve, Hyper-V, VirtualBox just lead the respective homepages, they are not relevant, but the KVM leads to important instructions. So on this page, some links should be followed, others not, but all links look the same. The same minishift-download link is linked twice, once on top,then on bottom left. So the structure of this page is a bit chaotic in my eyes.

The link to «installation document» leads to https://docs.openshift.org/latest/minishift/getting-started/installing.html, which is better structured, so the page https://www.openshift.org/minishift/ is simply an unnecessary duplicate.

My suggestion: On page https://www.openshift.org/minishift/ remove the «Get Started» part, and in the «Resources» part replace «Download» by «Installation» with link directly to https://docs.openshift.org/latest/minishift/getting-started. This would improve the navigation.

On page https://docs.openshift.org/latest/minishift/getting-started/installing.html, everything is understandable and clear. But at the end of the page, I missed some information, where I should continue reading. After a while, I found my position in the left menu and clicked the next item, Minishift Quickstart.

My suggestion: At the end of page, add a link to the next page,something like: «Next step: Minishift Quickstart» with link to https://docs.openshift.org/latest/minishift/getting-started/quickstart.html

In general: Since you have a paged documentation, a «back» and «next» link on every page would help.

Hope that helps. ;)

**Comment by mwaeckerlin** [comment]
@gbraad, regarding the LANG, I have a Ubuntu 16.04 installed from server ISO with KDE desktop and Swiss-German language environment. `LANG` is set to `de_CH.UTF-8`.

As you see above, `LANG= minishift start` is sufficient to dispose the language settings.

**Comment by gbraad** [comment]
@mwaeckerlin I noticed the `LANG=`, but perhaps this is acted on differently in Ubuntu. Needs some investigation.

But yeah, i can agree that the `--vm-driver` stuff is hidden away. It is mentioned on https://docs.openshift.org/latest/minishift/getting-started/quickstart.html#starting-minishift and tucked away in a note. If you can have a look at the PR I provided earlier... maybe it provides what you look for? If not, we can try to change it with your input. @robin-owen, please have a look at the comments in the above message.

**Comment by gbraad** [comment]
Thanks for the detailed description.

Note: I also want to protect my team of course... hope you understand. ;-). Although we are quite thick-skinned, we might take this back home or out onto the family

**Comment by robin-owen** [comment]
@mwaeckerlin @gbraad Some of the described issues should be taken care of by #2025. The primary intent behind that PR is to make navigation of the required steps much simpler. There's a viewable build of the documentation as it exists in that PR which you can access via the [CentOS CI artifacts](http://artifacts.ci.centos.org/minishift/minishift/docs/ondemand/robin-owen-issue-1817/openshift-origin/latest/minishift/getting-started/index.html).

In particular, that PR should help to address where to go next following each step of the installation process. Please let me know if there's anything more which we can be doing in that PR to help. :)

Also, @gbraad, I hope that the information in that PR also makes `--vm-driver` and the persistent `vm-driver` setting more visible, particularly for VirtualBox users.

**Comment by nalbam** [comment]
Thank you.
My system language was Korean. Change to English and work normally.
(LANG=ko_KR -> en_US)

**Comment by pilhuhn** [comment]
@gbraad Just for the record I ran into this on RHEL 7.4 with 

[17:19:05]  <pilhuhn>	$ echo $LANG
[17:19:05]  <pilhuhn>	de_DE.UTF-8

setting `LANG=C` before starting minishift helped.

**Comment by LalatenduMohanty** [comment]
@mwaeckerlin The artifacts generated from the PR to fix this issue is at http://artifacts.ci.centos.org/minishift/minishift/pr/2202/. Please try these and let us if solves the issue for you.

**Comment by LalatenduMohanty** [comment]
We need to fix https://github.com/openshift-cs/openshift.org/tree/master/source/minishift as it is the source code of https://www.openshift.org/minishift/ to address some of the comments mentioned by @mwaeckerlin.

**Comment by budhrg** [comment]
> https://www.openshift.org/minishift/ to address some of the comments mentioned ...

+1 for improving it @LalatenduMohanty since we always recommend this URL. This seems to be an entry point for most common users.

**Comment by slinkydeveloper** [comment]
I have this bug with my italian locale: `LANG=it_IT.UTF-8`
The workaround of running minishift commands prepending `LANG= ` works
